ankura/pipeline.py

```diff
@@ -614,15 +615,44 @@ def build_docwords(corpus, V=None):
 
 
 def test_train_split(corpus, num_train=None, num_test=None, **kwargs):
-    if not num_train:
+    if not num_train and not num_test:
         num_train = int(len(corpus.documents) * .8)
-    if not num_test:
+        num_test = len(corpus) - num_train
+    elif not num_train:
+        num_train = len(corpus.documents) - num_test
+    elif not num_test:
         num_test = len(corpus.documents) - num_train
 
-    doc_ids = np.random.permutation(len(corpus.documents))
-    train_ids, test_ids = doc_ids[:num_train], doc_ids[num_train: num_train+num_test]
-    train = Corpus([corpus.documents[d] for d in train_ids], corpus.vocabulary, corpus.metadata)
-    test = Corpus([corpus.documents[d] for d in test_ids], corpus.vocabulary, corpus.metadata)
+    print("Splitting test and train")
```

Comment:
This is garbo ![bitmoji](https://render.bitstrips.com/v2/cpanel/4db15a69-d173-4a80-90ba-7189280af0dd-58e1cf7a-21f2-4c43-82cb-72edb2944d01-v1.png?transparent=1&palette=1&width=246)

Reply:
![bitmoji](https://render.bitstrips.com/v2/cpanel/d74a88ff-6478-46db-af7d-55b121d9b2cf-b30ef521-856b-4209-82ef-10378d6686fd-v1.png?transparent=1&palette=1&width=246)